src/components/battles/animations/shared/Fainting.ts

```diff
@@ -45,17 +45,29 @@ export class Fainting<TGameStartr extends FullScreenPokemon> extends GeneralComp
                 const playerName = this.gameStarter.itemsHolder.getItem("name");
                 const partyIsWipedText: (string | string[])[][] = [[pokemon.nickname, " fainted!"]];
 
+                if (teamName === "opponent") {
```

Comment:
Similar to the comment above: if you have a very large block in an `if` and a very large block in an `else`, that's a sign it would be better to split it into two functions for readability.